doc/tutorial.rst

```diff
@@ -89,117 +97,140 @@ We will use the following very simple Dockerfile:
+Build Charliecloud image
+------------------------
 
-     $ sudo docker pull centos:8
+The three arguments here are the :code:`ch-image` subcommand :code:`build`,
+the option to enable unprivileged build workarounds :code:`--force`, and the
+context directory :code:`.`, which in this case is the current directory.
 
-   There are various resources and scripts online to help automate this
-   process, as well as :code:`misc/docker-clean.sh`.
+::
 
-Build Docker image
-------------------
+  $ ch-image build --force .
+  inferred image name: hello
+  2 FROM centos:8
+  will use --force: rhel8: CentOS/RHEL 8+
+  [...]
+  7 COPY ['.'] -> 'hello'
+  9 RUN ['/bin/sh', '-c', 'touch /usr/bin/ch-ssh']
+  --force: init OK & modified 1 RUN instructions
+  grown in 4 instructions: hello
 
-Charliecloud provides a convenience wrapper :code:`ch-build` around
-:code:`docker build` that mitigates some of the latter's more irritating
-characteristics. In particular, it passes through any HTTP proxy variables,
-and by default it uses the Dockerfile in the current directory, rather than at
-the root of the Docker context directory. (We will address the context
-directory in more detail later.)
+.. note::
+   :code:`ch-image` prints information about the build process while in
+   progress. While not demonstrated above, yellow is used for :code:`ch-image`
+   internal chatter.
 
-The two arguments here are a tag for the Docker image and the context
-directory, which in this case is the current directory.
+This image and the :code:`centos:8` base image used to build it are now
+visible in Charliecloud's builder storage:
 
 ::
 
-   $ ch-build -t hello .
-   Sending build context to Docker daemon  5.632kB
-   Step 1/4 : FROM centos:8
-    ---> 0d120b6ccaa8
-   [...]
-   Step 4/4 : RUN touch /usr/bin/ch-ssh
-    ---> eb5f42d5bb54
-   Successfully built eb5f42d5bb54
-   Successfully tagged hello:latest
-
-Note that Docker prints each step of the Dockerfile as it's executed.
-
-:code:`ch-build` and many other Charliecloud commands wrap various privileged
-:code:`docker` commands. Thus, you will be prompted for a password to escalate
-as needed. Note however that most configurations of :code:`sudo` don't require
-a password on every invocation, so privileged commands may be running even if
-you're not prompted for a password.
-
-Share image and other standard Docker stuff
+  $ ch-image list
+  centos:8
+  hello
+
+.. note::
+   Charliecloud uses image :code:`name:tag` to refer to images instead
+   of a computed hash like some other image builders.
+
+Sharing images
 -------------------------------------------
 
-If needed, the Docker image can be manipulated with standard Docker commands.
-In particular, image sharing using a public or private Docker Hub repository
-can be very useful.
+Charliecloud images in builder storage are just directories and can be
+exported as SquashFS filesystems or tar archives via :code:`ch-convert`.
+
+SquashFS:
```

Comment:
I think it would be better to do the tutorial with just one workflow, and support the other with a single sidebar or something saying how to translate.